### DISCUSSION TITLE
Tasmota Version on MQTT

### DISCUSSION THREAD

**Discussion by smhmaqsood** [arendst/Tasmota#11425] [Q&A] [answered]
Hi,
I'm trying to access the firmware version of tasmota through MQTT. But I'm unable to find it. Other options like "Toggle" & "Power" are working fine but unable to find the command for getting the firmware version through mqtt. Is there any such command ? if there is no such command then how can we make one?

**Comment by barbudor** [comment, marked as answer]
`status 2` response includes these details
```
{"StatusFWR":{"Version":"9.3.1(tasmota)","BuildDateTime":"2021-02-26T17:24:47","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8285","CR":"491/699"}}
```

**Reply by smhmaqsood** [reply]
Thanks

**Reply by sfromis** [reply]
The version is also included in the /INFO1 payload published automatically when Tasmota connects to MQTT.
`{"Module":"Generic","Version":"9.3.1.1(tasmota)","FallbackTopic": ....`

**Reply by smhmaqsood** [reply]
> 
> 
> `status 2` response includes these details
> 
> ```
> {"StatusFWR":{"Version":"9.3.1(tasmota)","BuildDateTime":"2021-02-26T17:24:47","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8285","CR":"491/699"}}
> ```

I'm using MQTT box and sending status like *cmnd/'my topic'/Status* and sending 2 as payload. and I've subscribed to *stat/'my topic'/Status* but I'm not getting any response. I've tried sending "2" as a payload in jason, string and decimal format. but still not getting any response. Am I doing something wrong? Because other MQTT commands are working perfectly

**Reply by barbudor** [reply]
Did you check in the device web  console what happened ?
Here is what I have for a same case, sending the command from mosquitto_pub command line tool:
```
08:09:12.299 MQT: stat/socket1/STATUS2 = {"StatusFWR":{"Version":"9.3.1(tasmota)","BuildDateTime":"2021-02-26T17:24:47","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8285","CR":"491/699"}}
```
the response is on `STATUS2`, not `STATUS`
Checking the console or having a MQTT tool that subscribe to "#" or "+/topic/#" and display all matching activity can be usefull to understand what happens.

**Reply by smhmaqsood** [reply]
Yes I did. But there is no response regarding "status" but other commands are working fine and can be seen in console.

Thanks For correcting me for response that I'll have to get the response at "stat/'my topic'/Status2" but am I working right with publisher?

**Reply by sfromis** [reply]
MQTT topics are case-sensitive. You should subscribe to STATUS2, not Status2.

**Reply by smhmaqsood** [reply]
Thanks for your guidance but still not working.

I'm publishing at *cmnd/'my topic'/STATUS* with *2* as payload. 
Is this a right way to do it?

**Reply by sfromis** [reply]
Don't get command and response mixed up. The thing about upper case was only for the subscribe, not for the command published to the Tasmota device. Tasmota will accept both upper and lower case in the command name, in the same way as it does in the console.

Watch the Tasmota console while doing your publish to send the status command to Tasmota. If you do get a `MQT:` message printed in the console, you should work on how you subscribe to get MQTT messages back.

If you do not, try activating `Weblog 3` and restart. The console will have messages about exactly what topics Tasmota subscribes to. You may have settings deviating from the default.

**Reply by smhmaqsood** [reply]
Thank you so much.
Got it. Its working now.
